shopinvader_backend_image_proxy/models/shopinvader_image_mixin.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com).
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class ShopinvaderImageMixin(models.AbstractModel):
+    _inherit = 'shopinvader.image.mixin'
+
+    @api.multi
+    def _get_image_data_for_record(self):
+        """
+        Inherit to replace the url if necessary
+        :return: list of dict
+        """
+        self.ensure_one()
+        result = super(
+            ShopinvaderImageMixin, self)._get_image_data_for_record()
+        # We have to ensure that the current recordset has a backend_id field
+        if 'backend_id' not in self._fields:
+            return result
+        if result and self.backend_id.image_proxy_url:
+            self._replace_image_url(result)
+        return result
+
+    @api.multi
+    def _replace_image_url(self, result):
```

Comment:
To simplify the code maybe it will be better to refactor the base module here : https://github.com/akretion/odoo-shopinvader/blob/10.0/shopinvader_image/models/shopinvader_image_mixin.py#L39
and add a method
```
def _prepare_resize_data(
                thumbnail = image_relation.image_id.get_or_create_thumbnail(
                    resize.size_x,
                    resize.size_y,
                    url_key=url_key)
                return  {
                    'src': thumbnail.url,
                    'alt': self.name,
}

image_data[resize.key] = self._prepare_data_resize(...)
```
So you just have to override the medthod prepare_data_resize here and avoid complex loop?

Reply:
@sebastienbeau Yes, easier like this! Thanks :)